frontend/src/components/molecules/Table.tsx

```diff
@@ -1,42 +1,99 @@
+  /** The function called to obtain the next elements */
+  nextFunction: (cursor: string) => Promise<any[]>;
 }
-
 /**
  * A Table component
  */
-const Table = ({ columns, rows }: TableProps) => {
-  return (
-    <DataGrid
-      columns={columns}
-      rows={rows}
-      sx={{ border: 0 }}
-      disableRowSelectionOnClick
-      initialState={{
-        pagination: {
-          paginationModel: {
-            pageSize: 100,
-          },
-        },
-      }}
-      pageSizeOptions={[]}
-      disableColumnMenu
+// initialCursor
+const Table = ({
+  columns,
+  rowData,
+  dataSetLength,
+  initialID,
+  nextFunction,
+}: TableProps) => {
+  const PAGE_SIZE = 5;
+  const [rows, setRows] = React.useState<any[]>(rowData);
+  const [cursor, setCursor] = React.useState<any[]>([initialID]);
+  const [lastElementID, setlastElementID] = React.useState<string>("");
+  const [paginationModel, setPaginationModel] = React.useState({
+    page: 0,
+    pageSize: PAGE_SIZE,
+  });
+  const [rowCountState, setRowCountState] = React.useState(dataSetLength);
+
+  const handlePaginationModelChange = async (
+    newPaginationModel: GridPaginationModel
+  ) => {
+    // stores the result from pagination request
+    var userData = [];
+    // tracks which page we on ["","userid1","userid2"...] --> [page1,page2,page3...]
+    var cursorArray = cursor;
 
-      // getRowId={(r) => r.DT_RowId}
-      // onCellClick={{console.log(getRowId)}}
-    />
+    if (paginationModel.page < newPaginationModel.page) {
+      console.log("going forwards");
+      cursorArray.push(lastElementID);
+      userData = await nextFunction(lastElementID);
+    } else {
+      console.log("going previous");
+      // one pop to get rid of last element
+      cursorArray.pop();
+      // second pop to get the details of last element
+      var cursor_after_pop = cursorArray.pop();
+      // push the cursor back on
+      cursorArray.push(cursor_after_pop);
+      userData = await nextFunction(cursor_after_pop);
+    }
+    adjustRows(userData);
+    setCursor(cursorArray);
+    setPaginationModel(newPaginationModel);
+  };
+
+  const adjustRows = async (rows: any[]) => {
+    var userData = [...rows];
+    console.log(userData);
+    const last_element_id = userData.pop().id;
+    console.log(last_element_id);
+    setRows(userData);
+    setlastElementID(last_element_id);
+  };
+  useEffect(() => {
+    adjustRows(rows);
+    console.log("HEREERE");
+  }, []);
+
+  console.log(rows);
+  console.log(cursor);
+  return (
+    <div>
+      {rows.length > 0 && (
+        <DataGrid
+          columns={columns}
+          rows={rows}
+          sx={{ border: 0 }}
+          disableRowSelectionOnClick
+          rowCount={rowCountState}
+          paginationModel={paginationModel}
+          onPaginationModelChange={handlePaginationModelChange}
+          pageSizeOptions={[]}
+          paginationMode="server"
+          disableColumnMenu
+        />
+      )}
+    </div>
   );
 };
```

Comment:
I'm assuming this PR is a WIP, there are a bunch of console logs which should be removed once code is finalized